File: server-springboot/src/main/java/com/campusbazaar/model/Notification.java
package com.campusbazaar.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document("notifications")
public class Notification {

    public enum Type { message, offer, offer_response, listing_expired, admin, report }

    @Id
    private String id;

    @Indexed
    private String user;

    private Type type;
    private String title;
    private String body;
    private String link;

    @Builder.Default
    @Indexed
    private boolean read = false;

    @CreatedDate
    private Instant createdAt;
}

File: server/models/Notification.js
const mongoose = require('mongoose');

const notificationSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    type: {
      type: String,
      enum: ['message', 'offer', 'offer_response', 'listing_expired', 'admin', 'report'],
      required: true
    },
    title: { type: String, required: true },
    body: { type: String, default: '' },
    link: { type: String, default: '' },
    read: { type: Boolean, default: false, index: true }
  },
  { timestamps: true }
);

module.exports = mongoose.model('Notification', notificationSchema);
